### PR TITLE
Add --local-only option

### PR DESCRIPTION
**Pull request by sripberger**
## Purpose of this pull request? 

- [ ] Documentation update
- [ ] Bug fix 
- [x] Enhancement
- [ ] Other, please explain:

## What changes did you make?

Added the `--local-only` CLI option, discussed [here](https://github.com/yeoman/environment/pull/113)

## Is there anything you'd like reviewers to focus on?

Nothing major except some changes to sinon usage in tests to match the new `env#lookup` behavior, where the callback arg might be the first or second arg.

### CODE REVIEW

File: lib/usage.txt [hunk 1]
```diff
@@ -7,6 +7,7 @@ General options:
   --no-color     # Disable colors
   --[no-]insight # Toggle anonymous tracking
   --generators	 # Print available generators
```
Comment (LitoMore) [inline lib/usage.txt]:
```suggestion
  --generators   # Print available generators
```

Comment (LitoMore) [reply lib/usage.txt]: Could you help fix this indentation? Looks there is a `\t` behind `--generators`.

Comment (sripberger) [reply lib/usage.txt]: It looks like this indentation irregularity was preventing tab completion for `--generators` from working at all. The completion tests omitted it, I assume for this reason.

Now that it's fixed I went ahead and added it to the test, as well. :+1:

**Comment by sripberger** [comment]
Honestly I'm not sure what's going on with the coverage checks. I just added a test that I thought would get rid of the problem but it doesn't. The coveralls UI is very difficult to follow so I'm not sure how to proceed.

Would be happy to make any other changes necessary.

**Comment by SBoudrias** [comment]
Ah no worries about the coverage. We're good to go.

Thanks for the PR.

**Pull request merged**